feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -440,64 +440,66 @@ class BillingWrapper(
         onSuccess: (Map<String, StoreTransaction>) -> Unit,
         onError: (PurchasesError) -> Unit
     ) {
-        withConnectedClient {
-            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE)
-
-            val querySubsPurchasesParams = BillingClient.ProductType.SUBS.buildQueryPurchasesParams()
-            if (querySubsPurchasesParams == null) {
-                onError(
-                    PurchasesError(
-                        PurchasesErrorCode.PurchaseInvalidError,
-                        PurchaseStrings.INVALID_PRODUCT_TYPE.format("queryPurchases")
-                    )
-                )
-                return@withConnectedClient
-            }
-            this.queryPurchasesAsyncWithTracking(
-                BillingClient.ProductType.SUBS,
-                querySubsPurchasesParams
-            ) querySubPurchasesAsync@{ activeSubsResult, activeSubsPurchases ->
-
-                if (!activeSubsResult.isSuccessful()) {
-                    val purchasesError = activeSubsResult.responseCode.billingResponseToPurchasesError(
-                        RestoreStrings.QUERYING_SUBS_ERROR.format(activeSubsResult.toHumanReadableDescription())
-                    )
-                    onError(purchasesError)
-                    return@querySubPurchasesAsync
-                }
-
-                val mapOfActiveSubscriptions =
-                    activeSubsPurchases.toMapOfGooglePurchaseWrapper(BillingClient.ProductType.SUBS)
+        executeRequestOnUIThread {
+            withConnectedClient {
```

Comment:
The only change here is wrapping the existing code inside the `executeRequestOnUIThread` method, which already has the logic to execute the block AFTER the billing client connection has been established. We already do this in all methods that interact with the billing client, except this one.